new_samples/main.go

```diff
@@ -0,0 +1,9 @@
+package main
+
+import "github.com/uber-common/cadence-samples/new_samples/worker"
+
+func main() {
+	for {
+		worker.StartWorker()
+	}
```

Comment:
probably want to actually run it, this creates an endless stream of
```
-486a-43e4-9af9-44f59658a6a0"}
2023-09-20T21:19:43.334-0500	INFO	worker/worker.go:50	Started Worker.	{"worker": "cadence-samples-worker"}
2023-09-20T21:19:43.339-0500	INFO	internal/internal_worker.go:834	Started Workflow Worker	{"Domain": "cadence-samples", "TaskList": "cadence-samples-worker", "WorkerID": "51336@stevenl-VLXY5H6XF7@cadence-samples-worker@8f25dab1-fc41-4448-9d1f-316ce10a8d04"}
2023-09-20T21:19:43.343-0500	INFO	internal/internal_worker.go:859	Started Activity Worker	{"Domain": "cadence-samples", "TaskList": "cadence-samples-worker", "WorkerID": "51336@stevenl-VLXY5H6XF7@cadence-samples-worker@8f25dab1-fc41-4448-9d1f-316ce10a8d04"}
2023-09-20T21:19:43.343-0500	INFO	worker/worker.go:50	Started Worker.	{"worker": "cadence-samples-worker"}
2023-09-20T21:19:43.349-0500	INFO	internal/internal_worker.go:834	Started Workflow Worker	{"Domain": "cadence-samples", "TaskList": "cadence-samples-worker", "WorkerID": "51336@stevenl-VLXY5H6XF7@cadence-samples-worker@a2a2ad8c-1c52-4026-8292-a2198db06ed4"}
```

from chat + a bit more searching: yeah, we just need the os signal strategy that fx uses.  theirs is correct.
https://github.com/uber-go/fx/blob/580de5816f8d027b371d7d02a2994d10754e19ce/docs/internal/apptest/run_test.go#L35-L39
```go
ch := make(chan os.Signal, 1)
signal.Notify(ch, os.Interrupt)
<-ch
```
there's a LOT of "listen to both interrupt and term", but Go's examples in signal.Notify don't show term, term doesn't appear to be necessary in my manual testing, and windows doesn't seem to have a term equivalent in Go.  so we just need interrupt.

Reply:
done to use a signal to interrupt